hs_core/tests/api/rest/test_unzip.py

```diff
@@ -0,0 +1,60 @@
+import os
+import tempfile
+import zipfile
+
+from rest_framework import status
+
+from hs_core.hydroshare import resource
+from hs_core.tests.api.utils import MyTemporaryUploadedFile
+
+from .base import HSRESTTestCase
+
+class TestPublicUnzipEndpoint(HSRESTTestCase):
+    def setUp(self):
+        super(TestPublicUnzipEndpoint, self).setUp()
+
+        self.tmp_dir = tempfile.mkdtemp()
+
+        # Make a text file
+        self.txt_file_name = 'text.txt'
+        self.txt_file_path = os.path.join(self.tmp_dir, self.txt_file_name)
+        txt = open(self.txt_file_path, 'w')
+        txt.write("Hello World\n")
+        txt.close()
+
+        self.raster_file_name = 'cea.tif'
+        self.raster_file_path = 'hs_core/tests/data/cea.tif'
+
+        # Make a zip file
+        zip_path = os.path.join(self.tmp_dir, 'test.zip')
+        with zipfile.ZipFile(zip_path, 'w') as zfile:
+            zfile.write(self.raster_file_path)
+            zfile.write(self.txt_file_path)
+
+        # Create a resource with zipfile, do not un-pack
+        payload = MyTemporaryUploadedFile(open(zip_path, 'rb'), name=zip_path,
+                                          content_type='application/zip',
+                                          size=os.stat(zip_path).st_size)
+
+        self.rtype = 'GenericResource'
+        self.title = 'My Test resource'
+        res = resource.create_resource(self.rtype,
+                                       self.user,
+                                       self.title,files=(payload,),
+                                       unpack_file=False)
+
+        self.pid = res.short_id
+        self.resources_to_delete.append(self.pid)
+        pass
+
+    def test_unzip(self):

```

Comment:
@pkdash Sure, what path should I use to test? Is there code somewhere that shows how to create a subdirectory in the resource?